src/rgw/rgw_bucket.cc

```diff
@@ -578,9 +580,10 @@ int rgw_remove_bucket(RGWRados *store, rgw_bucket& bucket, bool delete_children)
 
   RGWObjVersionTracker objv_tracker;
 
-  ret = store->delete_bucket(info, objv_tracker);
+  ret = store->delete_bucket(info, objv_tracker, false);
```

Comment:
i worry about the `delete_children=false` case here, but it looks like the `if (!objs.empty() && !delete_children) {` line above should have already caught that?

Reply:
if you don't pass the false, delete_bucket will detect the bad b-i entries and error out.

Reply:
(I should say it will detect the non-emptiness and error out)

Reply:
I see what you're saying now; I'll run that case to be sure.

Reply:
I checked and made a small change. If delete_children is false, we will still allow delete_bucket to check for children. The end-result is the same either way, but this will act as insurance against some subset of future changes.